### PR TITLE
Mashup: Reddit / Firebase Multiplayer Memory Game

### PR DESCRIPTION
**Pull request by djrosenbaum**

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -1,8 +1,756 @@
...
+            visibility: hidden;
+        }
+
+    </style>
+
   </head>
   <body>
+
+    <div class='memory'>
+        <div class='game_name'><span style="color:#c00;">Match Maker</span> - <span style="color:#c00;">Match Maker</span> - Make Me a Match</div>
```
Comment (afeld) [inline index.html]: Why not do these colors with CSS?

File: index.html [hunk 2]
```diff
@@ -1,8 +1,756 @@
...
+        .game_board :not(.match).memory_button:hover {
+            border: 1px solid #000;
+        }
+
+        .pressed {
+            background-color: #fff!important;
+            border-top: 1px solid #666!important;
+            border-right: 1px solid #ccc!important;
+            border-bottom: 1px solid #ccc!important;
+            border-left: 1px solid #666!important;
```
Comment (afeld) [inline index.html]: Eeek – try avoiding `!important` unless absolutely necessary.

File: index.html [hunk 3]
```diff
@@ -1,8 +1,756 @@
...
+    }
+    else if ( memory.cards.total_cards % 4 === 0) {
+        return 4;
+    }
+    else if ( memory.cards.total_cards % 3 === 0) {
+        return 3;
+    }
+    else if ( memory.cards.total_cards % 2 === 0) {
+        return 2;
+    }
```
Comment (afeld) [inline index.html]: How could you refactor this method to make it less repetitive?

File: index.html [hunk 4]
```diff
@@ -1,8 +1,756 @@
...
+            memory_button_html = '<div class="memory_button" data-location="' + i + '" data-number="' + memory.cards.card_array[i] + '"><img class="hidden" src="' + memory.reddit.imgur_array[memory.cards.card_array[i]] + '" width="125" height="125"></div>';
+            j = 1;
+        }
+
+        //DEAL CARD TO THE GAME BOARD
+        $('.game_board').append(memory_button_html);
+    }
+
+    memory.game_play.add_listeners();
+};
```
Comment (afeld) [inline index.html]: How could this method be made easier to read?

File: index.html [hunk 5]
```diff
@@ -1,8 +1,756 @@
...
+
+memory.game_board.game_won = function() {
```
Comment (afeld) [inline index.html]: I would make this method name indicate whether it's checking if the game is won, or if it's something that should be called _after_ the game is won.

**Comment by afeld** [comment]
My overall feedback is:
- Split CSS/JS into external files that are logically grouped.
- Make your methods shorter – see http://robots.thoughtbot.com/sandi-metz-rules-for-developers.
- Use OOP. Your code organization was good already, but this would encapsulate details that different parts of the application don't need to know about.

Really great work!

**Comment by djrosenbaum** [comment]
Updated Feedback by splitting files, writing more concise methods, and introducing an Object Oriented approach through reusable Playing_card and Card_deck constructors.

**Pull request closed without merging**